src/crate/client/cursor.py

```diff
@@ -49,9 +50,47 @@ def execute(self, sql, parameters=None, bulk_parameters=None):
 
         self._result = self.connection.client.sql(sql, parameters,
                                                   bulk_parameters)
-        if "rows" in self._result:
+
+        if "rows" not in self._result:
+            return
+
+        if "col_types" in self._result:
+            self.rows = iter(self._transform_result_types())
+
+        else:
             self.rows = iter(self._result["rows"])
 
+    def _transform_result_types(self):
+        """
+        Generate row items with column values converted to their corresponding
+        native Python types, based on information from `col_types`.
+
+        Currently, only converting to native `datetime` objects is implemented.
+        """
+        datetime_column_types = [11, 15]
```

Comment:
Does someone have a resource at hand where those column code types are enumerated? The table at [^1] coincidentally lists `_timestamp without time zone` as `1115`, but here the code is apparently expecting two-digit integer numbers.

I will be happy to receive further pointers for better educating myself on this topic.

[^1]: https://crate.io/docs/crate/reference/en/5.0/interfaces/postgres.html#pg-type

Reply:
https://crate.io/docs/crate/reference/en/5.0/interfaces/http.html#column-types


The code here will probably also need to handle arrays of timestamps and timestamps contained within objects.

Reply:
Thanks. I will add the conversion for container structures as well.

Reading up on your reference, I am asking myself whether it is appropriate that `type=15` (Unchecked object) is handled here as well?